Assets/Scripts/Utilities/DayCycleManager.cs

```diff
@@ -61,34 +67,64 @@ public static void RegisterEndOfDayCallback( EndOfDayCallback callback )
 
 	public static void DeregisterEndOfDayCallback( EndOfDayCallback callback )
 	{
-		instance._endOfDayCallback += callback;
+		instance._endOfDayCallback -= callback;
 	}
 
 	void StartMidnightOverlay()
 	{
-		_hasStartedMidnightOverlay = true;
 		StartCoroutine( FadeInMidnightOverlay() );
 	}
 
-	void EndMidnightOverlay()
+	void StartMorningOverlay()
 	{
-		_hasStartedMidnightOverlay = false;
-
-		StopCoroutine( _midnightOverlayCoroutine );
-
-		Color overlayColor = _midnightOverlay.color;
-		overlayColor.a = 0.0f;
-		_midnightOverlay.color = overlayColor;
+		StartCoroutine( FadeOutMidnightOverlay() );
 	}
 
+	/**
+	 * Used to fade to black at midnight.
+	 */
 	IEnumerator FadeInMidnightOverlay()
 	{
+		float elapsedTime = 0.0f;
+		while ( true )
+		{
+			if ( elapsedTime < _blackoutDuration )
```

Comment:
Consider reformatting to 

    elapsedTime += Time.deltaTime;
    Color overlayColor = _midnightOverlay.color;
    overlayColor.a = elapsedTime / _blackoutDuration;
    _midnightOverlay.color = overlayColor;

    if ( elapsedTime >= _blackoutDuration )
    {
        yield return break;
    }

    yield return null;